components/form/LifecycleHooks.vue

```diff
@@ -0,0 +1,80 @@
+<script>
+import Vue from 'vue';
+
+import HookOption from '@/components/form/HookOption';
+import { _VIEW } from '@/config/query-params';
+
+export default {
+  components: { HookOption },
+
+  props: {
+    mode: {
+      type:     String,
+      required: true,
+    },
+
+    // container spec
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+  },
+
+  data() {
+    const { postStart, preStop } = this.value;
+
+    return {
+      postStart, preStop, hookOptions: ['postStart', 'preStop']
+    };
+  },
+
+  computed: {
+    isView() {
+      return this.mode === _VIEW;
+    },
+  },
+
+  methods: {
+    update() {
+      const out = {
+        postStart: this.postStart,
+        preStop:   this.preStop
+      };
+
+      for (const prop in out) {
+        const val = out[prop];
+
+        if (val === '' || typeof val === 'undefined' || val === null || val === {}) {
+          Vue.delete(this.value, prop);
+        } else {
+          Vue.set(this.value, prop, val);
+        }
+      }
+
+      this.$emit('input', this.value);
+    },
+  }
+};
+</script>
+
+<template>
+  <div @input="update">
```

Comment:
It looks like the update function isn't being run when radio buttons are toggled, just when the string/number inputs are used. You can capture those radio changes and update after them if you move this to the HookOption components: 
`<HookOption v-model="postStart" :mode="mode" @input="update" />`

Reply:
Moved the `@input="update"` to each `HookOption`. Value is now updated when the radio buttons are toggled.